easybuild/tools/filetools.py

```diff
@@ -303,11 +303,27 @@ def remove_dir(path):
         dry_run_msg("directory %s removed" % path, silent=build_option('silent'))
         return
 
-    try:
-        if os.path.exists(path):
-            rmtree2(path)
-    except OSError as err:
-        raise EasyBuildError("Failed to remove directory %s: %s", path, err)
+    if os.path.exists(path):
+        ok = False
+        errors = []
+        # Try multiple times to cater for temporary failures on e.g. NFS mounted paths
+        max_attempts = 3
```

Comment:
make this an argument with default value 3?

Reply:
@boegel said to use `remove_dir`. And this function uses the max_attempts as an implementation detail which isn't changeable by the user. I'd rather keep this behavior